diff -Naur PyQt5_gpl-5.6/sip/QtGui/qguiapplication.sip PyQt5_gpl-5.6_patched/sip/QtGui/qguiapplication.sip
--- PyQt5_gpl-5.6/sip/QtGui/qguiapplication.sip	2016-04-24 12:55:17.000000000 +0200
+++ PyQt5_gpl-5.6_patched/sip/QtGui/qguiapplication.sip	2017-03-15 17:02:02.740245716 +0100
@@ -297,7 +297,9 @@
 pyqt5_qtgui_from_argv_list_t pyqt5_qtgui_from_argv_list;
 pyqt5_qtgui_update_argv_list_t pyqt5_qtgui_update_argv_list;
 
+%End
 // Forward declarations not in any header files but are part of the API.
+%If (WS_MACX)
 void qt_set_sequence_auto_mnemonic(bool enable);
 %End
 
diff -Naur PyQt5_gpl-5.6/sip/QtGui/qkeysequence.sip PyQt5_gpl-5.6_patched/sip/QtGui/qkeysequence.sip
--- PyQt5_gpl-5.6/sip/QtGui/qkeysequence.sip	2016-04-24 12:55:17.000000000 +0200
+++ PyQt5_gpl-5.6_patched/sip/QtGui/qkeysequence.sip	2017-03-15 17:02:37.573418556 +0100
@@ -243,4 +243,6 @@
 
 QDataStream &operator<<(QDataStream &in, const QKeySequence &ks /Constrained/) /ReleaseGIL/;
 QDataStream &operator>>(QDataStream &out, QKeySequence &ks /Constrained/) /ReleaseGIL/;
+%If (WS_MACX)
 void qt_set_sequence_auto_mnemonic(bool b);
+%End
